### PR TITLE
Make the CAPath option available on all OSes and change to throw on runtime on non-linux.

### PR DESCRIPTION
**Pull request by ahsonkhan**

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]

**Review by antkmsft — COMMENTED** [review]
Please add a test that verifies the behavior on Linux and non-Linux.

File: sdk/core/azure-core/src/http/curl/curl.cpp [hunk 1]
```diff
@@ -2319,9 +2315,14 @@ CurlConnection::CurlConnection(
     }
   }
 
-#if defined(AZ_PLATFORM_LINUX)
   if (!options.CAPath.empty())
   {
+#if defined(AZ_PLATFORM_LINUX)
```
Comment (LarryOsterman) [inline sdk/core/azure-core/src/http/curl/curl.cpp:2320]: This should just rely on the curl error being returned.

**Review by LarryOsterman — APPROVED** [review]

File: sdk/core/azure-core/src/http/curl/curl.cpp (hunk 1 above)
Comment (ahsonkhan) [reply sdk/core/azure-core/src/http/curl/curl.cpp:2320]: Agreed. I plan to remove this ifdef and if curl correctly handles an unsupported CA Path passed in correctly.

**Comment by antkmsft** [comment]
Let's get this one done tomorrow!

**Pull request merged**